packages/mermaid/src/diagrams/sankey/sankeyDB.ts

```diff
@@ -0,0 +1,105 @@
+import * as configApi from '../../config.js';
+import common from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  getAccDescription,
+  setAccDescription,
+  setDiagramTitle,
+  getDiagramTitle,
+  clear as commonClear,
+} from '../../commonDb.js';
+
+// Variables where graph data is stored
+// Sankey diagram represented by nodes and links between those nodes
+// We have to track nodes uniqueness (by ID), thats why we need hash also
+//
+let links: Array<SankeyLink> = [];
+let nodes: Array<SankeyNode> = [];
+let nodesHash: Record<string, SankeyNode> = {};
+let nodeAlign = 'justify';
+
+const setNodeAlign = function (alignment: string): void {
+  const nodeAlignments: string[] = ['left', 'right', 'center', 'justify'];
+  if (nodeAlignments.includes(alignment)) {
+    nodeAlign = alignment;
+  }
+};
+
+const getNodeAlign = () => nodeAlign;
+
+const clear = function () {
+  links = [];
+  nodes = [];
+  nodesHash = {};
+  nodeAlign = 'justify';
+  commonClear();
+};
+
+class SankeyLink {
+  constructor(public source: SankeyNode, public target: SankeyNode, public value: number = 0) {}
+}
+
+/**
+ * @param source - Node where the link starts
+ * @param target - Node where the link ends
+ * @param value - number, float or integer, describes the amount to be passed
+ */
+const addLink = function (source: SankeyNode, target: SankeyNode, value: number): SankeyLink {
+  const link: SankeyLink = new SankeyLink(source, target, value);
+
+  links.push(link);
+
+  return link;
+};
+
+class SankeyNode {
+  constructor(public ID: string, public label: string = ID) {}
```

Comment:
SankeyNode is never created with a label, so do we need it?

Reply:
This is for the further development, but if you prefer to keep it clean and add only that features that are in use, I'll remove it

Reply:
No need, if you have immediate plans, keep it :)

Reply:
Decided to keep it clean